test/test_params_case_translator.rb

```diff
@@ -86,4 +86,20 @@ def test_data_case_camel_complex
     assert response_params['meta']['tag_list'].first.has_key? 'display_name'
     assert response_params['links'].has_key? 'user_groups'
   end
+
+  def test_type_should_always_be_snake_case
+    params = { 'data' => { 'type' => 'dataElement' } }
+    env = mock_env(
+      '/',
+      'POST',
+      {
+        'HTTP_X_API_DATA_CASE' => 'camel-lower',
+        'action_dispatch.request.request_parameters' => params
+      }
+    )
+    status, headers, body = @translator.call(env)
+    assert_equal 200, status
+    response_params = body.first
+    assert_equal 'data_element', response_params['data']['type']
+end
```

Comment:
Is this an extra `end`?

I'd also like to see more comprehensive tests for this so we can guard against unexpected behavior. Perhaps additional payloads that have relationships and a kay with a value that has keys and values that shouldn't be touched by the case translator.